src/Raven.Client/Documents/Changes/DatabaseChanges.cs

```diff
@@ -545,8 +545,8 @@ private async Task DoWork(string nodeTag)
                         _url = new Uri($"{_serverNode.Url}/databases/{_database}/changes"
                             .ToLower()
                             .ToWebSocketPath(), UriKind.Absolute);
-
-                        await _client.ConnectAsync(_url, _cts.Token).ConfigureAwait(false);
+                        var timeoutCts = CancellationTokenSource.CreateLinkedTokenSource(new[] { _cts.Token, new CancellationTokenSource(TimeSpan.FromSeconds(15)).Token });
```

Comment:
That would timeout in 15 seconds, no? 
Not just if this is hanged

Reply:
>That would timeout in 15 seconds
yes - the 15 sec token is only on the `ConnectAsync` call tho (not other calls)
what is the concern?